multiversx_sdk/core/transaction_computer.py

```diff
@@ -32,10 +35,20 @@ def compute_transaction_fee(self, transaction: ITransaction, network_config: INe
         return int(fee_for_move + processing_fee)
 
     def compute_bytes_for_signing(self, transaction: ITransaction) -> bytes:
+        self._ensure_fields(transaction)
+
+        if transaction.version >= 2 and self.has_options_set_for_hash_signing(transaction):
+            return self.compute_hash_for_signing(transaction)
```

Comment:
We've removed this logic from JS.

https://github.com/multiversx/mx-sdk-js-core/pull/416

Reply:
Indeed. Removed here as well.